src/views/v2/Bridge/AmountInput/index.tsx

```diff
@@ -51,19 +56,17 @@ const DebouncedTextField = memo(
 
     const onInnerChange: ChangeEventHandler<HTMLInputElement> = useCallback(
       (e) => {
-        let value = e.target.value;
-        if (value === '.') value = '0.';
+        const value = removeCommas(e.target.value);
 
-        const numValue = Number(value);
-
-        if (isNaN(numValue) || numValue < 0) {
-          // allows all but negative numbers
+        if (!isValidDecimalInput(value)) {
           return;
         }
 
-        setInnerValue(e.target.value);
-        onChange(e.target.value); // callback with no delay
-        deferredOnChange(e.target.value);
+        const formattedValue = formatWithCommas(value);
+
+        setInnerValue(formattedValue);
```

Comment:
@emreboga 
Kind of curious, why we have 2 setStates that track each other. We already set the value in state, innerValue can just be derived directly? Which means, all we need to do is change line 88 to be `formatWithCommans(value)` and innerValue can just be deleted?

Is it due to the focus issue since there is a guard for it? But if a user is focused and typing, it should update what they type, no?

In any case @kyleleighton we probably should not set UI-formatted values in state, only raw data. How that is represented visually, should be at the render level.